src/Receipt/Receipt.php

```diff
@@ -88,35 +71,61 @@ private function setupReceipt() {
 	 * @since 2.7.0
 	 */
 	public function getDetailGroupObject( $class ) {
+		$classNames           = $this->getDetailGroupList();
+		$detailGroupClassName = $classNames[ array_search( $class, $classNames, true ) ];
+
+		$object = new $detailGroupClassName( $this->donationId );
+
 		/**
-		 * Use this filter to handle object initialization for your detail group class.
+		 * fire the action for receipt detail group.
 		 *
 		 * @since 2.7.0
 		 */
-		$object = apply_filters( 'give_receipt_create_detail_group_object', null, $class, $this->donationId );
+		 do_action( 'give_new_receipt_detail_group', $object );
```

Comment:
So is the idea, here, that a `DetailGroup` subclass isn't constructed until it's retrieved? And is this something an addon will be using?

Reply:
@JasonTheAdams by using this hook and `DetailGroup ` subclass object they can easily add a new detail item to the group as I am using it in FFM pr: https://github.com/impress-org/give-form-field-manager/pull/312